fbw-common/src/wasm/systems/systems/src/payload/mod.rs

```diff
@@ -34,11 +36,398 @@ pub struct CargoInfo<'a> {
     pub payload_id: &'a str,
 }
 
+pub trait NumberOfPassengers {
+    fn number_of_passengers(&self, ps: usize) -> i8;
+}
+
+pub trait PassengerPayload {
+    fn total_passenger_load(&self) -> Mass;
+    fn total_target_passenger_load(&self) -> Mass;
+
+    fn center_of_gravity(&self) -> Vector3<f64>;
+    fn fore_aft_center_of_gravity(&self) -> f64;
+    fn target_center_of_gravity(&self) -> Vector3<f64>;
+    fn target_fore_aft_center_of_gravity(&self) -> f64;
+}
+
+pub trait CargoPayload {
+    fn total_cargo_load(&self) -> Mass;
+    fn total_target_cargo_load(&self) -> Mass;
+
+    fn center_of_gravity(&self) -> Vector3<f64>;
+    fn fore_aft_center_of_gravity(&self) -> f64;
+    fn target_center_of_gravity(&self) -> Vector3<f64>;
+    fn target_fore_aft_center_of_gravity(&self) -> f64;
+}
+
+#[derive(Debug)]
+pub struct BoardingAgent<const P: usize> {
+    order: [usize; P],
+}
+impl<const P: usize> BoardingAgent<P> {
+    pub fn new(order: [usize; P]) -> Self {
+        BoardingAgent { order }
+    }
+
+    pub fn handle_one_pax(&self, pax: &mut [Pax; P]) {
+        for ps in self.order {
+            if pax[ps].pax_is_target() {
+                continue;
+            }
+            pax[ps].move_one_pax();
+            break;
+        }
+    }
+}
+
+#[derive(Debug)]
+pub struct PassengerDeck<const N: usize, const G: usize> {
+    pax: [Pax; N],
+    boarding_agents: [BoardingAgent<N>; G],
+
+    boarding_gates_id: VariableIdentifier,
+    boarding_gates: usize,
+}
+impl<const N: usize, const G: usize> PassengerDeck<N, G> {
+    pub fn new(
+        context: &mut InitContext,
+        pax: [Pax; N],
+        boarding_agents: [BoardingAgent<N>; G],
+    ) -> Self {
+        PassengerDeck {
+            pax,
+            boarding_agents,
+            boarding_gates_id: context.get_identifier("NUM_BOARDING_GATES".to_owned()),
+            boarding_gates: 0,
+        }
+    }
+
+    fn num_stations(&self) -> usize {
+        self.pax.len()
+    }
+
+    fn pax_num(&self, ps: usize) -> i8 {
+        self.pax[ps].pax_num() as i8
+    }
+
+    fn total_pax_num(&self) -> i32 {
+        (0..self.num_stations())
+            .map(|ps| self.pax_num(ps) as i32)
+            .sum()
+    }
```

Comment:
I think this violates the principle of least knowledge/law of demeter since this method should not require awareness of the method ``pax_num()`` from ``Pax``